javascript/javascript2/week2/movies_task.js

```diff
@@ -0,0 +1,138 @@
+// So if there were 3 movies that contained Surfer, 1 with Alien and 2 with Benjamin, you would return 6.
+// Can you make sure the search is case insensitive?
+
+const keyWord1 = 'Surfer'.toLowerCase();
+const keyWord2 = 'Alien'.toLowerCase();
+const keyWord3 = 'Benjamin'.toLowerCase();
+
+const sumMoviesWithKeyWords = movies.filter(movie => {
+    const titleWords = movie.title.toLowerCase().split(' ');
+    return (titleWords.includes(keyWord1) || titleWords.includes(keyWord2) || titleWords.includes(keyWord3))
+}) ;
+console.log(sumMoviesWithKeyWords);
+
+// 7.Create an array of movies where a word in the title is duplicated. Fx "Star Wars: The Clone Wars" the word Wars is duplicated.
+// Here are some madeup examples of movies with duplicated words in the title: "The three men and the pistol", "Chase three - The final chase"
+
+const duplicatedWordMovies = movies.filter(movie => {
+    const titleWords = movie.title.toLowerCase().split(' ');
+    return hasDuplicateWords(titleWords);
+})
+
+function hasDuplicateWords(words) {
+    for (let i = 0; i < words.length - 1; i++) {
+        for (let j = i + 1; j < words.length; j++) {
+            if (words[i] === words[j]) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+console.log(duplicatedWordMovies);
+
+// 8.Calculate the average rating of all the movies using reduce.
+
+function sumRating(previousValue, el) {
+    return previousValue + el.rating;
+}
+
+const rating = movies.reduce(sumRating, 0);
+const averageRating = rating / movies.length;
+
+console.log(averageRating);
+
+// 9.Count the total number of Good, Average and Bad movies using reduce.
+// A return could fx be {goodMovies: 33, averageMovies: 45, badMovies: 123}
+
+const goodMoviesCount = movieRatings.reduce((previousValue, el) => {
+    if (el.tag === 'Good') {
+        return previousValue + 1;
+    } else {
+        return previousValue;
+    }
+}, 0);
+
+console.log(goodMoviesCount);
+
+const averageMoviesCount = movieRatings.reduce((previousValue, el) => {
+    if (el.tag === 'Average') {
+        return previousValue + 1;
+    } else {
+        return previousValue;
+    }
+}, 0);
+
+console.log(averageMoviesCount);
+
+const badMoviesCount = movieRatings.reduce((previousValue, el) => {
+    if (el.tag === 'Bad') {
+        return previousValue + 1;
+    } else {
+        return previousValue;
+    }
+}, 0)
+
+console.log(badMoviesCount);
+
+const moviesCount = movieRatings.reduce(
+    (previousValue, el) => {
+        if (el.tag === 'Good') {
+            previousValue.goodMoviesCount++;
+        } else if (el.tag === 'Average') {
+            previousValue.avarageMoviesCount++;
+        } else {
+            previousValue.badMoviesCount++;
+        }
+        return previousValue;
+    },
+    {
+        goodMoviesCount : 0,
+        avarageMoviesCount : 0,
+        badMoviesCount : 0
+    }
+);
+console.log(moviesCount);
```

Comment:
here its seems like you have written more code to find out the good movies ,average movies,bad movies.

i think below code is enough if you made some changes.

const moviesCount = movieRatings.reduce(
    (previousValue, el) => {
        if (el.tag === 'Good') {
            previousValue.goodMoviesCount++;
        } else if (el.tag === 'Average') {
            previousValue.avarageMoviesCount++;
        } else {
            previousValue.badMoviesCount++;
        }
        return previousValue;
    },
    {
        goodMoviesCount : 0,
        avarageMoviesCount : 0,
        badMoviesCount : 0
    }
);
console.log(moviesCount);

is it makes sense